### ISSUE TITLE
Version for 4.1

### ISSUE THREAD

**Issue by Bjoern-Ge** [KQMATH/moodle-mod_capquiz#214]
Dear CAPQuiz-Team,
on moodle 3.9 I was using CAPQuiz. 
Are there any planes for an update, working on 4.1?
By now, the taps "Questions" ans "Class list" are not working at all.
For the students, it is working.

<img width="658" alt="CAPQuiz" src="https://user-images.githubusercontent.com/93242256/221219476-3fd6cd81-808a-4d34-93d8-5349bd072424.png">

**Comment by hgeorgsch** [comment]
Yes, there is, and I am really sorry that we have not been able to release yet.
It is possible that the moodle4 branch already works on Moodle 4.x, but I have not found the time to test it properly.

**Comment by Bjoern-Ge** [comment]
That sounds good. I have the version from moodle.org. 
Is that branch available? I haven't seen it. If you like, I can test it on 4.1

**Comment by hgeorgsch** [comment]
It should be if you use git, but since you ask, you probably don't :-)
If you go to the code page, there is a drop-down to select the branch (default is development).  Here you can select moodle4.  Then the «Code» button allows you to download.  The normal approach in my circles is to copy and paste the command to use git, but there is also a link for «Download ZIP» which I have never tried.

Would be very useful if you test indeed.

**Comment by Bjoern-Ge** [comment]
In the moodle4 version used on moodle 4.1, the question list is fixed. 
The Class list is still an empty page for old CAPQuises. 
When adding a new quiz, the class list can be used.

**Comment by hgeorgsch** [comment]
Sorry for the delay @Bjoern-Ge .
As far as I can tell, this is working for new Quizzes, but the upgrade from Moodle 3 does not work.  
Is that a correct?  I don't know if you still pay attention to this.

I wonder if I should make a release, moving forward on Moodle 4.
I have no idea how easy it will be to make working upgrade scripts.  I have never done that before, and therefore every input is welcome, while I try to research it ...

If @andstor has some ideas if he still pays attention?

**Comment by Bjoern-Ge** [comment]
@hgeorgsch yes, in moodle 4.1 it was working, only the old Quizzes were not working.
I think making a release like this is OK without the update script. 
I also tested with a student's account. With real students, I haven't tested it yet.

**Comment by hgeorgsch** [comment]
I am unable to reproduce the problem with the classlist.
In my case I start with v0.6.3 on Moodle 3.9 and upgrade to the now current development branch of CAPQuiz on Moodle 4.0.  This appears to work as it should.  I see no recent changes to explain why my test would be different from that of @Bjoern-Ge .

I will move on to test with 4.1 and 4.2.
